src/tcp_server.rs

```diff
@@ -129,6 +123,9 @@ impl TcpServer {
                                             ClientMessage::ChangeLayer { new } => {
                                                 kanata.lock().change_layer(new);
                                             }
+                                            ClientMessage::RequestLayerNames {} => {
+                                                kanata.lock().tcp_send_layer_names_requested = true;
```

Comment:
It seems simpler to me to remove all of the code changes in `kanata/mod.rs` and form+send the message here. Could you explain why you chose to do the changes in the processing loop? Perhaps I'm missing something.

The reason I can think of, is that the current code follows the existing pattern used to send to all clients. This message does not need to be sent to all clients though, only the one requesting.

I also don't foresee this code running frequently, so to me it is fine to execute more code while the lock is held (only sufficiently long to create the vec of layer names) and then form the message here.

Reply:
83b74b2bf6f9b60749851f0ee87138a4dcaa6177